### PR TITLE
Handle runner not found

### PR DESCRIPTION
**Pull request by ericsciple**

### CODE REVIEW

**Comment by pje** [comment]
So is the intention: "404s should never be retryable from the runner's perspective"?

File: src/Sdk/WebApi/WebApi/BrokerHttpClient.cs [hunk 1]
```diff
@@ -115,6 +115,7 @@ public async Task<TaskAgentMessage> GetRunnerMessageAsync(
             {
                 switch (brokerError.ErrorKind)
                 {
+                    case BrokerErrorKind.RunnerNotFound:
```
Comment (luketomlinson) [inline src/Sdk/WebApi/WebApi/BrokerHttpClient.cs:118]: I wouldn't re-use the same AccessDeniedException with ErrorCode=1. That means the runner deprecation error in a few other places. Can we throw another error here instead? I don't really care what, but probably should be different than the Runner version being too old?

Comment (ericsciple) [reply src/Sdk/WebApi/WebApi/BrokerHttpClient.cs:118]: Oh interesting. I thought this was a generic mechanism to bubble out. Will look again.

**Comment by ericsciple** [comment]
> So is the intention: "404s should never be retryable from the runner's perspective"?

Yes if the runner doesn't exist anymore, then shut down

**Review by luketomlinson — APPROVED** [review]

**Pull request merged**